File: Python/stronghold/hamm.py
# Counting Point Mutations
import sys


def calc_hamming_distance(s1, s2):
    distance = 0
    for c1, c2 in zip(s1, s2):
        if c1 != c2:
            distance += 1
    return distance


def main():
    reader = sys.stdin
    s1 = next(reader).strip()
    s2 = next(reader).strip()
    result = calc_hamming_distance(s1, s2)
    print(result)


if __name__ == '__main__':
    main()

File: Scala/src/main/scala/Stronghold/HAMM.scala
package Stronghold

object HAMM {
  def calcHammingDistance(s1: String, s2: String): Int =
    s1.lazyZip(s2).count{ case (c1, c2) => c1 != c2 }

  def main(args: Array[String]): Unit = {
    val reader: Iterator[String] = scala.io.Source.stdin.getLines()
    val s1: String = reader.next()
    val s2: String = reader.next()
    val result: Int = calcHammingDistance(s1, s2)
    println(result)
  }
}
